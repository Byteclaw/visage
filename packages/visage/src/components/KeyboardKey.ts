import { createComponent } from '../core';

export const KeyboardKeyStyles: VisageStyleSheet = {
  backgroundColor: 'shades',
  borderRadius: 'keyboardBorderRadius',
  boxShadow:
    '0 0 0 1px color(shades if(isDark color(shades tint(15%)) color(shades shade(15%)))), 0 3px 0 0 shades, 0 3px 0 1px color(shades if(isDark color(shades tint(15%)) color(shades shade(15%))))',
  color: 'shadesText',
  display: 'inline-block',
  fontFamily: 'monospace',
  fontSize: 'inherit',
  lineHeight: 'inherit',
  m: 0,
  p: 0,
  px: 0.5,
  textAlign: 'center',
  textTransform: 'lowercase',
  userSelect: 'none',
  whiteSpace: 'nowrap',
};

export const KeyboardKey = createComponent('kbd', {
  displayName: 'KeyboardKey',
  styles: KeyboardKeyStyles,
});
