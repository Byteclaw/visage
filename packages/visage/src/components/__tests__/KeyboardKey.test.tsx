import React from 'react';
import { KeyboardKey } from '../KeyboardKey';
import { createTestTheme, render } from './render';

describe('KeyboardKey', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<KeyboardKey />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by KeyboardKey face', () => {
    const { asFragment } = render(<KeyboardKey />, {
      ds: {
        theme: createTestTheme({
          KeyboardKey: {
            borderRadius: '10px',
          },
        }),
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
