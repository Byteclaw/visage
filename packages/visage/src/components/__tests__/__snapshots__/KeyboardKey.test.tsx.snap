// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardKey is extendable by KeyboardKey face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  background-color: rgb(204,204,204);
  border-radius: 10px;
  box-shadow: 0 0 0 1px rgb(173,173,173),0 3px 0 0 rgb(204,204,204),0 3px 0 1px rgb(173,173,173);
  color: rgb(255,255,255);
  display: inline-block;
  font-family: monospace;
  font-size: inherit;
  font-weight: 400;
  line-height: inherit;
  margin: 0px;
  padding: 0px;
  padding-left: 4px;
  padding-right: 4px;
  text-align: center;
  text-transform: lowercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
}

<kbd
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`KeyboardKey renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  background-color: rgb(204,204,204);
  border-radius: keyboardBorderRadius;
  box-shadow: 0 0 0 1px rgb(173,173,173),0 3px 0 0 rgb(204,204,204),0 3px 0 1px rgb(173,173,173);
  color: rgb(255,255,255);
  display: inline-block;
  font-family: monospace;
  font-size: inherit;
  font-weight: 400;
  line-height: inherit;
  margin: 0px;
  padding: 0px;
  padding-left: 4px;
  padding-right: 4px;
  text-align: center;
  text-transform: lowercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
}

<kbd
    class="emotion-0"
  />
</DocumentFragment>
`;
